From 8e29feef2f5c8fd48dbbd738784e504cba02c950 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 19 Feb 2020 13:11:31 +0100
Subject: [PATCH 1/3] tmpfiles: create MYSQL_DATADIR

This is a no-op if the directory exists, but makes sure it is created by
systemd-tmpfiles with proper permissions otherwise.

This solves packaging issues when the user MYSQLD_USER is created by
systemd-sysusers and uid is not known in advance.

Also this now sets the No_COW attribute.
---
 support-files/tmpfiles.conf.in | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/support-files/tmpfiles.conf.in b/support-files/tmpfiles.conf.in
index e41585c08a8..d28ff43d0d9 100644
--- a/support-files/tmpfiles.conf.in
+++ b/support-files/tmpfiles.conf.in
@@ -7,6 +7,10 @@
 # * innodb_temp_data_file_path
 # * rocksdb_tmpdir
 # etc.
-# It shouldn't be used for datadir which is why it
-# may be disabled.
 @DISABLE_MYSQL_UNIX_ADDRDIR@d @MYSQL_UNIX_ADDRDIR@ 0755 @MYSQLD_USER@ root -
+
+# Create the data directory
+# Upstream thinks this is a bad idea, mainly because the mechanism has "tmp"
+# in its name. We do it anyway - no-op if exists, and works just fine. :)
+d @MYSQL_DATADIR@ 0700 @MYSQLD_USER@ @MYSQLD_USER@ -
+h @MYSQL_DATADIR@ - - - - +C

From c84b97b4d1a9b37784bb631dbba2b4acfdf65eb7 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Wed, 1 Jan 2025 22:30:45 +0100
Subject: [PATCH 2/3] sysusers: create a fully locked system account

https://github.com/systemd/systemd/blob/v257/NEWS#L767-L777
https://www.freedesktop.org/software/systemd/man/latest/sysusers.d.html#u
---
 support-files/sysusers.conf.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/support-files/sysusers.conf.in b/support-files/sysusers.conf.in
index a975b29476a..e2d1608bc6e 100644
--- a/support-files/sysusers.conf.in
+++ b/support-files/sysusers.conf.in
@@ -1 +1 @@
-u @MYSQLD_USER@ - "MariaDB" @MYSQL_DATADIR@
+u! @MYSQLD_USER@ - "MariaDB" @MYSQL_DATADIR@

From 2df22e4dbac2be8b3b2a033412cb5d7548cdc926 Mon Sep 17 00:00:00 2001
From: Christian Hesse <mail@eworm.de>
Date: Mon, 18 May 2026 13:20:26 +0200
Subject: [PATCH 3/3] scripts: make aware of being a package

---
 scripts/mysql_install_db.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/mysql_install_db.sh b/scripts/mysql_install_db.sh
index 5049e833e92..5ce1886f22e 100644
--- a/scripts/mysql_install_db.sh
+++ b/scripts/mysql_install_db.sh
@@ -35,7 +35,7 @@ silent_startup="--silent-startup"
 log_error=""
 
 force=0
-in_rpm=0
+in_rpm=1
 ip_only=0
 cross_bootstrap=0
 do_resolve=0
